### ISSUE TITLE
Mega console spam due to depreciation 

### ISSUE THREAD

**Issue by Stono** [swagger-api/swagger-js#1143]
Hey, 
This is from running `npm run test`.

```
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
      ✓ should resolve references as deeply as possible (116ms)
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
      ✓ should handle this weird case
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
      ✓ should not stop if one $ref has error
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
      ✓ should include fullPath in invalid $ref type
      ✓ should be able to overwrite fetchJSON
    deeply resolved
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
jsonpatch.apply is deprecated, please use `applyPatch` for applying patch sequences, or `applyOperation` to apply individual operations.
```

Seems to be coming from `fast-json-patch`.
I tried updating it to the latest minor of v1 but it didn't helps, and updating it to 2.x.x caused a whole bunch of tests to fail.

Honestly, looking at your dependencies, quite a few are severely out of date. 

```
 deep-extend                                 ^0.4.1  →   ^0.5.0
 fast-json-patch                             ~1.1.8  →   ~2.0.5
 isomorphic-form-data                         0.0.1  →    1.0.0
 lodash                                      4.16.2  →   4.17.4
 babel-eslint                                ^6.0.2  →   ^8.0.1
 babel-loader                                ^6.3.2  →   ^7.1.2
 babel-plugin-transform-object-rest-spread   6.16.0  →   6.26.0
 babel-plugin-transform-runtime              6.15.0  →   6.23.0
 eslint                                     ^3.18.0  →   ^4.8.0
 eslint-config-airbnb-base                  ^11.1.1  →  ^12.0.1
 expect                                     ^1.20.2  →  ^21.2.1
 license-checker                             ^8.0.3  →  ^14.0.0
 mocha                                       ^2.4.5  →   ^3.5.3
 webpack                                     ^2.2.1  →   ^3.6.0
```

You're going to get more of these sorts of problems unless you keep on top of those!

**Comment by Stono** [comment]
I would really appreciate this being merged and released as at the moment all users of a CLI on our project get a full terminal of console spam...

**Comment by Stono** [comment]
As a work around for others who have bumped into this, as `swagger-js` is loose on the patch version `~1.1.8`, by doing `npm install --save fast-json-patch@1.1.8`, npm will just use that version, rather than the `1.1.10` which has this warning in.

**Comment by shockey** [comment]
@Stono thanks for the heads up - I'm looking over this now!

**Comment by shockey** [comment]
For context: we originally used `fast-json-patch@^1`, but when they released their 1.2 series, we had some issues (see https://github.com/Starcounter-Jack/JSON-Patch/issues/178).

**Comment by shockey** [comment]
RE out of date dependencies - yep, we are falling behind. The core team is focused on building out OpenAPI 3.0 functionality in our JavaScript projects, so refactoring third-party interfaces in order to keep up is not at the top of our list. But thank you for bringing it up 😄 if you have the time, we'd appreciate a PR that brings us up to speed.

I just merged your PR, and am cutting a release now. It should be live on NPM in about 15 minutes. Thanks again for reporting this.

Closing!
